### DISCUSSION TITLE
Getting values from advanced module fields for PDF template

### DISCUSSION THREAD

**Discussion by bfeddes** [YetiForceCompany/YetiForceCRM#16895] [How to] [answered]
Hey everyone, I am currently working on creating a PDF template for Quotes. However, I am unable to get data from the advanced module section, I can only get the data from the **Fields and Blocks** section. For example, when I type $(record : subject)$ I am able to get the subject printed on the PDF from the Quote. However, if I type $(record : name)$ to get the item name in the Advanced Module tab, it does not work. I was wondering if anyone knew the solution to this.

I was also wondering if there is a way to have the PDF print each product that is in the quote, so if there are 3 products, it can print the three product names, prices, quantities, etc.

Thanks everyone.

**Comment by mariuszkrzaczkowski** [comment, marked as answer]
Try this 
https://github.com/YetiForceCompany/YetiForceCRM/blob/1116b3b9cdd0ff68f5b1addb5fd134a007d80801/app/TextParser.php#L43

https://github.com/YetiForceCompany/YetiForceCRM/issues/11476
https://github.com/YetiForceCompany/YetiForceCRM/issues/12056

**Reply by bfeddes** [reply]
Worked like a charm! Thank you!
